### DISCUSSION TITLE
New Unguided Level - Stealth mission

### DISCUSSION THREAD

**Discussion by Moburma** [swfans/syndwarsfx#188] [Show and tell]
Here's another entirely original level I've been working on. Not got a name or story concept for it, other than it would be the equivalent of the two other Reykjavik levels, where you're infiltrating the C3 base to get items that lead to storyline revelations. 
In this case though I went for a different angle and rather than a dangerous firefight, this one is about using a character with minimal weapons to use stealth to get through the base, kill some scientists, and then get out safely.

The stealth mechanics aren't used much by the game, there's a De Saxo level that plays with them, but I think that's it. They actually work quite well, aside from it needing a bit of trial and error for the player to understand the visual range of enemies (not that far, actually), so this is exploring an underused side of the game.

Anyway, your mission, should you choose to accept it, is to enter the base, recover a briefcase one from scientist, and kill another (I might give him a briefcase to get as well, not done it yet). One important thing to note is there is  a clone shield near the first scientist, this will be **required** to progress through the level. There's no objective for it at the moment, but possibly I would put one in to stop any confusion on how important it is - it would definitely be mentioned in any briefing though.

Complete the above, and then get back to the IML.

This is again a work in progress, I think it peters out a bit once you've got the briefcase, so any ideas to keep the challenge up would be welcome. Of course also any thoughts on what is good/bad, and even just plain is this going in a good direction or not (keep/scrap!) would be great. I learnt quite a lot working on this one, and actually changed my mind on how the level should work after I realised it was much more fun to NOT start with the clone shield (in fact this is an issue - would it be more fun to NOT get it at all and have to do pure stealth?).

Download is [here](https://filebin.net/7o4p50juxx6hc5wq), and add this to your miss002.ini for Mission5:

[mission5]
Name = MISSNO5
SourceID = 0
MapNo = 79
LevelNo = 4

[missuccess5]
P00 = GAME_OBJ_GET_ITEM( Thing(-270,1133) )
P01 = GAME_OBJ_P_DEAD( Thing(126,1017) )
P02 = GAME_OBJ_ALL_G_ARRIVES( Group(0), Coord(30848,0,2304), Radius(10) )

Start with -m 2,5

**Comment by mefistotelis** [comment]
> The stealth mechanics [...] it needing a bit of trial and error for the player to understand the visual range of enemies (not that far, actually), so this is exploring an underused side of the game.

I was actually thinking about the Scanner weapon from first Syndicate. It was pretty useless, giving only a zoomed out radar. We could add similar thing to SW, but instead of radar it could scan the nearby CHIPs - providing the player with either walk path of enemies or line of sight circles. Or both.

That would eliminate some of the trial-and-error frustration.

**Comment by Moburma** [comment]
Yeah, I think that's a workable idea, the only thing is it wouldn't be that useful 95% of the time - it's rare you ever need or want to be that stealthy. Not sure how best to offer such a thing - would be a good improvement to something like the useless infrared mode!

I made a video of the level here: https://www.youtube.com/watch?v=7G6PecN0MWc

Took a LOT of attempts to get the above footage, I think the difficulty is actually quite good now, hard but not impossible, actually challenging, and mostly I died from rushing things or being greedy and instantly regretting it. The pseudo random nature of Wander mode also results in it subtly changing in an interesting way I hadn't foreseen.
Not sure if it actually needs anything else added, could do with someone else to playtest it and give their opinions. There are a couple of exploits in some areas I noticed to make a couple of bits trivial, so might slip in a few more enemies to block them off.

**Comment by geist22** [comment]
Damn, this is pretty hard. I'm not exactly a master of stealth missions, so for me this was a bit frustrating, particularly the first part. I'd say the patrols in the area with the Clone Shield are a bit too tight with too little leeway. Also, one of the militia there frequently gets stuck walking against walls inside the building, and when you finally find a time when the two nearby agents' patrols overlap in a way to let you move (to 1:25 in your videa), he sees you 4 out of 5 times, and him attacking will push you out from behind the cover into the agent's sight. I guess this isn't intended?

![swars](https://github.com/user-attachments/assets/4c36283e-e6fb-471c-aef8-2dcdb4967d0a)

After failing a number of attempts trying your route and only reaching the Clone Shield once, I've found success rushing to the wall on the other side while the agent was away and hiding behind the two silos, but the timing is extremely strict, you can just barely make it.

What I don't really like is that your Unguided has the... stamina? equivalent of an agent with zero mods, so you can only run for a few seconds before you need to find a place and wait for it to regenerate so you can run again. For all her faults, De Saxo can at least run indefinitely (judging from her persuadertron range, she has a lv3 brain).
I don't think the Clone Shield needs an objective, the way it's placed makes it pretty obvious that this is something you want. The second part feels fine to me, there's no need to make missions more complicated than they need to be. When I finally cleared the first part and made it to the second scientist, I entered the area with the Clone Shield and wanted to kill the militia first, so the scientist escaped to the previous area and I died getting there again 🤣. Got him the second time I've made it this far, while the militia was just staring at a wall as I killed his buddy. The first part of the mission is already tricky enough, so it's fine for the second part to be relatively easy. If the second part was equally difficult, having to redo the first part every time you fail in the later stages of the mission will quickly lead to frustration.

I've never really considered missions with De Saxo (or Wu) stealth missions. Enemies attack you on sight just as they always do, and since De Saxo dies in like 2 bullets from a Minigun, leaving enemies alive and squeezing through between them is just an unnecessary risk, doubly so since you're easily capable of clearing out all enemies in the levels with the LR Rifle. From my experience, the only upside those special characters have over your usual agents is that the police won't attack them. I've always found it both strange and annoying that the police in SWARS is basically just another enemy faction that will attack your agents but not enemies unless scripted or accidentally hit.

I do think having an actual stealth mission is neat and a welcome change of pace, even if I prefer combat heavy missions myself (my favorites are Phoenix, Rome (both sides), and I always fully clear out Colombo (yes, the entire map) rather than using the Clone Shield).

**Reply by Moburma** [reply]
> Damn, this is pretty hard. I'm not exactly a master of stealth missions, so for me this was a bit frustrating, particularly the first part. I'd say the patrols in the area with the Clone Shield are a bit too tight with too little leeway. Also, one of the militia there frequently gets stuck walking against walls inside the building, and when you finally find a time when the two nearby agents' patrols overlap in a way to let you move (to 1:25 in your videa), he sees you 4 out of 5 times, and him attacking will push you out from behind the cover into the agent's sight. I guess this isn't intended?

I think that's fair, particularly the agent near the Scientist is a bit annoying/much I might remove him, or at least make him have a longer pause time at each extreme so he stays out of the way longer. He's particularly annoying when you go to actually kill the scientist. I suppose also this would help with something else - I thought of having him or some other agents react when the scientist dies (just walking near to where his body would be) as it always seems a bit weird this guy is now dead and no one notices. If he's not there, it makes more sense there is no reaction!

Also full disclosure - I forgot to mention another mechanic I put in. Most of the soldiers will stop what they are doing and follow you if you get to close to them. The idea is they find you suspicious when you have the clone shield on. This means that if you are shielded and walk close to them you will inevitably get into combat with them. Possibly this is a bit too much, it's probably hard enough already.  It also causes problems with the enemy you identified above. He's just set to wander, but he tends to get stuck inside the bunker, as you pointed out. I don't think this is bad _per se_ as it deprives you of an easy way to get to the objective and makes you think on your feet, BUT a) you may not see this happen and therefore not know he is there (the map file seems to determine which buildings can/can't be seen inside with no level override which is annoying) which is unfair, and b) He can "see" you through the walls because of the above follow mechanic! So you can hug the wall there to get past other enemies only to unwittingly set him on your trail... I might just remove him, he's a bit too erratic and annoying.

> 
> What I don't really like is that your Unguided has the... stamina? equivalent of an agent with zero mods, so you can only run for a few seconds before you need to find a place and wait for it to regenerate so you can run again. For all her faults, De Saxo can at least run indefinitely (judging from her persuadertron range, she has a lv3 brain). 

Yeah, I'm torn on this. It seems from the existing levels that probably it was meant to be this way, with the unguided being slow and very vulnerable, but so too later on it gives you characters that don't get fatigued in this way (see Tale of Two Cities). On balance I think it IS an annoying thing and should be done away with quickly throughout the main campaign and perhaps just kept for a few levels. I actually like it on this level because it ups the stakes and means you need to plan ahead. That said it is annoying at times and undeniably takes ages to recover fully. Not decided what to do, I'm just using "default" (there isn't really such a thing, I'm just using common early game values) values for stamina, might boost them up so it still happens but not as severe, or just remove. I do lean more towards keeping but toning down. 

This is something interesting for the campaign as a whole, the level "Talks Ambush" has a chase that is quite thrilling because of this, but at the same time also feels clumsy/annoying. It's not clear if it was intended to be like this or the game stats/mechanics changed and the abandoned levels just ended up unadjusted. Certainly towards the end of the campaign it's just cheap artificial difficulty and should not be used. That said how could you explain it away, they've all been going to the gym more!?

>I entered the area with the Clone Shield and wanted to kill the militia first, so the scientist escaped to the previous area and I died getting there again 🤣. Got him the second time I've made it this far, while the militia was just staring at a wall as I killed his buddy. The first part of the mission is already tricky enough, so it's fine for the second part to be relatively easy. If the second part was equally difficult, having to redo the first part every time you fail in the later stages of the mission will quickly lead to frustration.

Haha, I cackled evilly reading this! This was a thing I put in ages ago and forgot about. It doesn't usually trigger as the scientist will fall back to the default flee reaction if you have a gun out. BUT one of my best runs when I was trying to record the video (it took me at least 40 attempts!) ended with it happening and I just turned the game off in despair. Who put this mean idea in! I think I will take it out, it's way too harsh and was from when the level was easier (no APC drop off).

> I've never really considered missions with De Saxo (or Wu) stealth missions. Enemies attack you on sight just as they always do, and since De Saxo dies in like 2 bullets from a Minigun, leaving enemies alive and squeezing through between them is just an unnecessary risk, doubly so since you're easily capable of clearing out all enemies in the levels with the LR Rifle. From my experience, the only upside those special characters have over your usual agents is that the police won't attack them. I've always found it both strange and annoying that the police in SWARS is basically just another enemy faction that will attack your agents but not enemies unless scripted or accidentally hit.

Yeah, I wouldn't say they are stealth missions _per se_, but some parts can be played that way. Most obviously the mission Soul Crusader I found it useful to outmanoeuvre some enemies first by sneaking into the base to isolate them, and actually played a lot of it as a proper stealth game in order to kill them more easily. That's what really gave me the idea to extend the stealth idea further as I knew it was possible.

oh, and definitely agree on the Police, they are used in weird and quite lazy ways a lot of the time. I always remember a quote from I think John Romero working on Doom that people always respond well to making enemies fight, it's fun and makes the player feel clever, of getting one over on the game. I need to do more experiments with Psycho Gas to that end, but from what I've seen so far it's pretty crappy unfortunately, an awesome idea on paper but quite feeble in practice. @mefistotelis pointed out that on the really awesome early Bangkok Unguided level it's setup deliberately so you can force a police/zealot conflict which is awesome as it not only works to your advantage but is great to watch unfold. I never even thought of this as they are so poorly used through most of the campaign, definitely a high bar for smart use of Police.

> I do think having an actual stealth mission is neat and a welcome change of pace, even if I prefer combat heavy missions myself (my favorites are Phoenix, Rome (both sides), and I always fully clear out Colombo (yes, the entire map) rather than using the Clone Shield).

Yeah, I don't think I would do another one in this style as it would get stale quickly (one thing I did think of though would be a similar mission but you have the Displacertron as a weapon and have to use it to strategically temporarily remove enemies. The problem is most maps aren't really linear enough to facilitate this).

**Comment by Moburma** [comment]
Here's a new version:
https://filebin.net/x7unblxyrvdh672i

Updates:

- Removed annoying "bunker" soldier
- Added two new stationary agents just to stop the player bypassing the corridor to the transport tubes and final scientist
- Extended the patrol wait times for the agent nearest the first scientist. This should mean the player has more time to kill the scientist before having to deal with him

**Comment by geist22** [comment]
Just to be clear, the part I've had the most trouble with before was moving between the two houses in front of the wall to the Clone Shield. The route you took in your video looked unintuitive to me, as your Unguided looks very exposed in that spot between the bunker and that tall, round building. What I first tried was to move along the wall on the other side, but the agent there sees you. You can make a run for it as he's moving away and hide behind one of the silos, the timing doesn't feel as tight as I first tought. I didn't have any trouble killing the scientist, as I moved into the bunker and waited for him to enter so he wouldn't run off into the other direction.

This new version is much better. With the militia gone, moving through the middle feels much less exposed, and with the bunker being a safe space now, your return trip is so much easier.

I thought it would make sense for the scientists to run away on sight as they sometimes do during the campaign, but this would be a nightmare balance-wise with all the enemies outside. Could be interesting if you had the option to use the barrels to your advantage, for example if the first scientist was stationary next to the barrels, having and agent left and right to his side, and you could pick up a Minigun inside the bunker and shoot a nearby barrel from behind the bunker to get them killed... but that wouldn't be very stealthy now, would it?

**Reply by Moburma** [reply]
The scientist actually does run away on sight, but only if you have a gun out. It's a group thing, I could set the scientist faction to the enemy of the player one, then they would always run away. The trouble is this also makes it more annoying in my opinion, as the short range of the uzi and the number of bullets it takes to kill the scientists means he can sometimes get away into the range of the agents and it all drags out longer. 

Definitely agreed on the barrels thing, I really wanted to include them somehow, but once you give people the minigun (which has the needed range) people will just try and kill everything (and probably succeed). Might be able to do something interesting with just the purple psycho gas barrels (i.e. get the agents shooting each other with them), but I haven't tried it. I did think of giving the agents much more powerful weapons (i.e. Plasma Lances), which would balance out starting with the minigun, but think it's enough as-is.

**Reply by mefistotelis** [reply]
Barrels - you may have an objective like "set up time ignition", which is really "GOTO LOCATION". After meeting this objective, a hidden person may start waiting for specific amount of turns, then self-destruct. This way you could have explosive results without setting up bombs, and do a larger delay than the usual bombs have. Not sure if this specific level should be further complicated by that though.

**Reply by Moburma** [reply]
Yeah, I didn't want to script anything, that would be simple to do but not in the spirit of what we meant here. What I mean is a load of the action in this level takes place near the barrels. Therefore they could give the player a serious advantage in a good way in a level based around avoiding combat if they are used to blow up the enemies. So what's the problem? I deliberately only gave the player an uzi to discourage combat (you're meant to sneak around). This means you cannot shoot the barrels and live! You can't escape the blast wave of the explosives with its limited range, making them impossible to use.

I could give the player a minigun, but that just encourages combat, defeating the point of the level. I might toy with doing this and see what it actually plays like.

**Reply by mefistotelis** [reply]
Giving the player any weapon would drastically change reception of the level.

Though we could make one level where the player has a weapon, but the level is divided into sections with passages protected by strong enemies attacking you if you have any weapon, no matter if hidden or not. This way you'd have to drop all weapons when crossing.

**Reply by Moburma** [reply]
That's actually a really great idea, I'd like to do some kind of unguided levels earlier in the campaign where they are really vulnerable/don't have the upper hand and you need sneaky stuff like this to progress. Plus again it's an aspect of the game that is not really explored much in the other campaigns - you're not going to want to dump your expensive weapon inventory when it persists between levels, after all.

**Comment by Moburma** [comment]
Ok, so I think this one is pretty finished as a level, and quite fun to play.  I will put the finishing touches on and add it to the campaign. I came up for a name for it: Access Violation (geddit?)

**Comment by marceloabx** [comment]
man, I can't thank you and all other developers enough. Is there a way to donate some money for this project?

**Reply by Moburma** [reply]
We couldn't take money from this project even if we wanted to - it's after all Bullfrog's game and code so would not be legal. The best way to show appreciation is to invest your time - If you can program, get involved. Even if you can't, spread the word, tell people about the project on social media, play test new levels and give feedback, log reports on any bugs you find. You can see only a few people know/take part in the project currently, so the more people we can find who are interested, the faster and better progress on the game can be.
